### PR TITLE
feat(tracing): introduce deal_id-based trace logs to particle execution

### PR DESCRIPTION
**Pull request by gurinderu**
## Description
Adds a span propagation with the additional deal_id tracing information.

## Motivation
Simplify the passing of the contextual information for debugging (i.e deal_id)

## Related Issue(s)
https://linear.app/fluence/issue/NET-493/small-span-demo

## Checklist
- [x] The code follows the project's coding conventions and style guidelines (https://doc.rust-lang.org/1.0.0/style/README.html).
- [x] All tests related to the changes have passed successfully.
- [x] Documentation has been updated to reflect the changes (if applicable).
- [x] All new and existing unit tests have passed.
- [x] I have self-reviewed my code and ensured its quality.
- [x] I have added/updated necessary comments to aid understanding.

## Reviewer Checklist
- [ ] Code has been reviewed for quality and adherence to guidelines.
- [ ] Tests have been reviewed and are sufficient to validate the changes.
- [ ] Documentation has been reviewed and is up to date.
- [ ] Any questions or concerns have been addressed.

### CODE REVIEW

File: aquamarine/src/actor.rs [hunk 1]
```diff
@@ -65,7 +67,9 @@ where
         functions: Functions<F>,
         current_peer_id: PeerId,
         key_pair: KeyPair,
+        deal_id: Option<String>,
     ) -> Self {
+        let span = tracing::info_span!("Actor", deal_id = deal_id);
```
Comment (folex) [inline aquamarine/src/actor.rs:72]: maybe pass just `Span`? Why pass `deal_id`?

File: aquamarine/src/actor.rs [hunk 2]
```diff
@@ -207,6 +215,7 @@ where
                 };
                 (reusables, res.effects, res.stats)
             }
+            .instrument(span)
```
Comment (folex) [inline aquamarine/src/actor.rs:218]: can we simplify it like that?
```suggestion
            .instrument(self.span.clone())
```

File: aquamarine/src/actor.rs [hunk 3]
```diff
@@ -195,6 +202,7 @@ where
         // TODO: get rid of this clone by recovering key_pair after `vm.execute` (not trivial to implement)
         let key_pair = self.key_pair.clone();
         // TODO: add timeout for execution https://github.com/fluencelabs/fluence/issues/1212
+        let span = self.span.clone();
```
Comment (folex) [inline aquamarine/src/actor.rs:205]:
```suggestion
```

File: aquamarine/src/actor.rs (hunk 1 above)
Comment (gurinderu) [reply aquamarine/src/actor.rs:72]: It was the main idea - to create a shared id for the execution sub graph

File: aquamarine/src/log.rs [hunk 4]
```diff
@@ -1,10 +1,38 @@
+use humantime::FormattedDuration;
+
 /// Function that prescribes different log levels for different builtin namespaces
-pub fn builtin_log_level(service: &str) -> log::Level {
+pub fn builtin_log_level(service: &str) -> fn((String, FormattedDuration, String)) -> () {
```
Comment (folex) [inline aquamarine/src/log.rs:4]: Why return a closure?

If you can argue on why, then gotta rename the function and change the comment

File: aquamarine/src/actor.rs (hunk 2 above)
Comment (gurinderu) [reply aquamarine/src/actor.rs]: done

File: aquamarine/src/log.rs (hunk 4 above)
Comment (gurinderu) [reply aquamarine/src/log.rs:4]: tracing macro doesn't support a dynamic value (level), so I have decided to fix that in the simplest way

Comment (gurinderu) [reply aquamarine/src/log.rs]: renamed

File: aquamarine/src/actor.rs (hunk 1 above)
Comment (folex) [reply aquamarine/src/actor.rs:72]: shared id is inside `Span`, right?

make plumber create `Span` and pass it to `Actor::new`. Then would be no need to pass `deal_id`

File: aquamarine/src/log.rs (hunk 4 above)
Comment (folex) [reply aquamarine/src/log.rs]: why doesn't it support? I don't understand. You can do that, right?
```
let level = builtin_log_level(service);
event!(level, "blablabla")
```
```

File: aquamarine/src/particle_executor.rs [hunk 5]
```diff
@@ -69,58 +69,61 @@ impl<RT: AquaRuntime> ParticleExecutor for RT {
         let (particle, calls) = p;
         let particle_id = particle.id.clone();
         let data_len = particle.data.len();
+        let span = tracing::info_span!("Execute");
         let task = tokio::task::Builder::new()
             .name(&format!("Particle {}", particle.id))
             .spawn_blocking(move || {
-                let now = Instant::now();
```
Comment (folex) [inline aquamarine/src/particle_executor.rs:75]: Maybe move insides of that closure into a separate function? too many indentation levels, I think

File: aquamarine/src/log.rs (hunk 4 above)
Comment (gurinderu) [reply aquamarine/src/log.rs]: no)

Comment (gurinderu) [reply aquamarine/src/log.rs]: Fixed

File: aquamarine/src/actor.rs (hunk 1 above)
Comment (gurinderu) [reply aquamarine/src/actor.rs]: Fixed

File: aquamarine/src/particle_executor.rs (hunk 5 above)
Comment (gurinderu) [reply aquamarine/src/particle_executor.rs:75]: Out of scope for the current pr

File: aquamarine/src/plumber.rs [hunk 6]
```diff
@@ -55,10 +55,6 @@ pub struct Plumber<RT: AquaRuntime, F> {
     key_manager: KeyManager,
 }
 
-unsafe impl<RT: AquaRuntime, F: ParticleFunctionStatic> Send for Plumber<RT, F> {}
-
-unsafe impl<RT: AquaRuntime, F: ParticleFunctionStatic> Sync for Plumber<RT, F> {}
```
Comment (folex) [inline aquamarine/src/plumber.rs:60]: how come it's not needed anymore?

**Review by folex — APPROVED** [review]

File: aquamarine/src/plumber.rs (hunk 6 above)
Comment (gurinderu) [reply aquamarine/src/plumber.rs:60]: I was committed by mistake

**Review by justprosh — APPROVED** [review]

**Review by kmd-fl — COMMENTED** [review]
I don't like how it looks. :c
Sad there's no other nicer way to do the tracing

**Pull request merged**